### PR TITLE
44 refactor attributemap

### PR DESCRIPTION
**Pull request by ryanlong1004**
AttributeMap was convoluted and good be removed.

### CODE REVIEW

File: tests/test_scheduler.py [hunk 1]
```diff
@@ -391,16 +386,16 @@ def test_lsf2():
     expected = """#BSUB -P account_name
 #BSUB -q batch
 #BSUB -W 00:01:00
-#BSUB -R affinity[core(1)]
+#BSUB -n 12
```
Comment (aerorahul) [inline tests/test_scheduler.py:389]: This test shouldn't be passing.
The number here should be 144, based on the `props` input.

Comment (ryanlong1004) [reply tests/test_scheduler.py:389]: created issue

File: tests/test_scheduler.py [hunk 2]
```diff
@@ -415,9 +410,9 @@ def test_lsf3():
     expected = """#BSUB -P account_name
 #BSUB -q batch
 #BSUB -W 00:01:00
-#BSUB -R affinity[core(1)]
+#BSUB -n 2
```
[the remark this answers is not in the document]
Comment (ryanlong1004) [reply tests/test_scheduler.py:413]: created issue #48

**Review by aerorahul — APPROVED** [review]
Looks good.
Issues have been created to address the LSF test incorrectness.

**Pull request merged**